### PR TITLE
ojopaul:feature/#64_Only_allow_one_author_mapped_to_each_user

### PR DESCRIPTION
**Pull request by ojopaul**
## Description

- Only allow one author mapped to each user #64

## Benefits

Form validation added to when creating and/or updating author terms to prevent multi user link for author profile

## Applicable issues

https://github.com/publishpress/PublishPress-Authors/issues/64

## Checklist

- [x] I have created a specific branch for this pull request before committing, starting off the current HEAD of `development` branch. 
- [x] I'm submitting to the `development`, feature/hotfix/release branch. (Do not submit to the master branch!)
- [x] This pull request relates to a specific problem (bug or improvement).
- [x] I have mentioned the issue number in the pull request description text.
- [x] All the issues mentioned in this pull request relate to the problem I'm trying to solve.
- [x] The code I'm sending follows the [PSR-12](https://www.php-fig.org/psr/psr-12/) coding style.

### CODE REVIEW

File: src/core/Classes/Admin_Ajax.php [hunk 1]
```diff
@@ -209,4 +209,36 @@ public static function handle_author_get_user_data()
...
+     * Handle a request to validate mapped author.
+     */
+    public static function handle_mapped_author_validation()
+    {
+
+        $response['status']  = 'success';
+        $response['content'] = esc_html__('Request status.', 'publishpress-authors');
+
+        //do not process request if nonce validation failed
+        if (empty($_POST['nonce']) || !wp_verify_nonce($_POST['nonce'], 'mapped_author_nonce')) {
```
Comment (andergmartins) [inline src/core/Classes/Admin_Ajax.php:223]: This is missing the `sanitize_key` for when calling `wp_verify_nonce`.

File: src/core/Classes/Author_Editor.php [hunk 2]
```diff
@@ -625,4 +626,28 @@ public static function admin_notices()
...
+
+    /**
+     * Author term mapped limit validation
+	 *
+	 * @param string|WP_Error $term     The term name to add, or a WP_Error object if there's an error.
+	 * @param string          $taxonomy Taxonomy slug.
+     * 
+     * @return array|WP_Error
+     */
+    public static function filter_pre_insert_term($term, $taxonomy)
```
Comment (andergmartins) [inline src/core/Classes/Author_Editor.php:638]: This is working well. But we need to make an additional verification for the Author URL field. We can't have two authors using the same Author URL, and we can't have an author using the Author URL of another user (not only author), if not set as mapped to that user.

**Review by andergmartins — CHANGES_REQUESTED** [review]
Working well, good job. Just need some additional adjustments.

File: src/core/Classes/Author_Editor.php (hunk 2 above)
Comment (ojopaul) [reply src/core/Classes/Author_Editor.php:638]: I'll only be checking for user with slug and skip term with same slug as the check is already existing. It's impossible to add author with same slug. So, will be adding a check for user

**Comment by ojopaul** [comment]
@andergmartins i've updated the check and improve code formatting as well. Will check my other PR too

**Review by andergmartins — APPROVED** [review]
Thanks @ojopaul it is working like a charm.

**Pull request merged**